beacon_node/execution_layer/src/engine_api/http.rs

```diff
@@ -625,40 +671,151 @@ impl HttpJsonRpc {
-                    Err(Error::ServerMessage {
-                        code: error.code,
-                        message: error.message,
-                    })
+            // Generate and add a jwt token to the header if auth is defined.
+            if let Some(auth) = &self.auth {
+                request = request.bearer_auth(auth.generate_token()?);
+            };
+
+            let start = Instant::now();
+            let body = request.send().await;
+            let elapsed = start.elapsed();
+
+            // open file and write the elapsed time in ms in a  new line
+            let mut file = OpenOptions::new()
+                .append(true)
+                .create(true)
+                .open("http_times.txt")
+                .unwrap();
+            writeln!(file, "{}", elapsed.as_millis()).unwrap();
+            drop(file);
+
+            let body: JsonResponseBody = body?.error_for_status()?.json().await?;
+
+            match (body.result, body.error) {
+                (result, None) => serde_json::from_value(result).map_err(Into::into),
+                (_, Some(error)) => {
+                    if error.message.contains(EIP155_ERROR_STR) {
+                        Err(Error::Eip155Failure)
+                    } else {
+                        Err(Error::ServerMessage {
+                            code: error.code,
+                            message: error.message,
+                        })
+                    }
                 }
             }
         }
+        else {
+
+            let mut wspackage = self.wspackage.lock().await;
+            wspackage.id += 1;  // id for next req
+
+            // initialize the ws connection if needed
+            match wspackage.status {
+                WsStatus::Uninitialized => {
+                    if let Some(auth) = &self.auth {
+                        let wsrouter = WsRouter::new_with_jwt(self.url.full.to_string(), auth.generate_token()?).await.map_err(|e| Error::WebsocketError(e.to_string()))?;
+                        wspackage.wsrouter = Some(wsrouter);
+                        wspackage.status = WsStatus::Initialized;
+                    }
+                    else {
+                        let wsrouter = WsRouter::new(self.url.full.to_string()).await.map_err(|e| Error::WebsocketError(e.to_string()))?;
+                        wspackage.wsrouter = Some(wsrouter);
+                        wspackage.status = WsStatus::Initialized;
+                    }
+                }
+                _ => {} // either already initialized or disabled
+            }
+
+            let body = JsonRequestBody {
+                jsonrpc: JSONRPC_VERSION,
+                method,
+                params,
+                id: json!(&wspackage.id),
+            };
+
+            let payload = serde_json::to_string(&body)?;
+          
+            let start = Instant::now();
+            let response = wspackage.wsrouter.as_ref().unwrap().send(payload.clone(), wspackage.id).await;
```

Comment:
You can avoid the clone here if you don't log the payload below

```suggestion
            let response = wspackage.wsrouter.as_ref().unwrap().send(payload, wspackage.id).await;
```